src/Fixer/Operator/BinaryOperatorSpacesFixer.php

```diff
@@ -30,65 +31,149 @@
 final class BinaryOperatorSpacesFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface
 {
     /**
-     * @var AbstractAlignFixerHelper[]
+     * @internal
      */
-    private $alignFixerHelpers = [];
+    const SINGLE_SPACE = 'single_space';
+
+    /**
+     * @var array
+     */
+    private static $defaultConfiguration = [
+        '=' => self::SINGLE_SPACE,
+        '*' => self::SINGLE_SPACE,
+        '/' => self::SINGLE_SPACE,
+        '%' => self::SINGLE_SPACE,
+        '<' => self::SINGLE_SPACE,
+        '>' => self::SINGLE_SPACE,
+        '|' => self::SINGLE_SPACE,
+        '^' => self::SINGLE_SPACE,
+        '+' => self::SINGLE_SPACE,
+        '-' => self::SINGLE_SPACE,
+        '&' => self::SINGLE_SPACE,
+        '&=' => self::SINGLE_SPACE,
+        '&&' => self::SINGLE_SPACE,
+        '||' => self::SINGLE_SPACE,
+        '.=' => self::SINGLE_SPACE,
+        '/=' => self::SINGLE_SPACE,
+        '=>' => self::SINGLE_SPACE,
+        '==' => self::SINGLE_SPACE,
+        '>=' => self::SINGLE_SPACE,
+        '===' => self::SINGLE_SPACE,
+        '!=' => self::SINGLE_SPACE,
+        '<>' => self::SINGLE_SPACE,
+        '!==' => self::SINGLE_SPACE,
+        '<=' => self::SINGLE_SPACE,
+        'and' => self::SINGLE_SPACE,
+        'or' => self::SINGLE_SPACE,
+        'xor' => self::SINGLE_SPACE,
+        '-=' => self::SINGLE_SPACE,
+        '%=' => self::SINGLE_SPACE,
+        '*=' => self::SINGLE_SPACE,
+        '|=' => self::SINGLE_SPACE,
+        '+=' => self::SINGLE_SPACE,
+        '<<' => self::SINGLE_SPACE,
+        '<<=' => self::SINGLE_SPACE,
+        '>>' => self::SINGLE_SPACE,
+        '>>=' => self::SINGLE_SPACE,
+        '^=' => self::SINGLE_SPACE,
+        '**' => self::SINGLE_SPACE,
+        '**=' => self::SINGLE_SPACE,
+        '<=>' => self::SINGLE_SPACE,
+        '??' => self::SINGLE_SPACE,
+    ];
+
+    /**
+     * @var array<string, string>
+     */
+    private $alignOperatorTokens = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        parent::configure($configuration);
+
+        $this->configuration = self::$defaultConfiguration;
+
+        if (0 === count($configuration)) {
+            return;
+        }
+
+        foreach ($configuration as $name => $value) {
+            if ('align_equals' === $name || 'align_double_arrow' === $name) {
+                $this->configuration = $this->resolveOldConfig($configuration);
+
+                break;
+            }
+
+            if ('<=>' === $name && !defined('T_SPACESHIP')) {
+                unset($this->configuration[$name]);
```

Comment:
That would be less convenient for project supporting multiple PHP versions as these cannot have one CS configuration. We do this at other places as well (`ReturnTypeDeclarationFixer` can be configured to on PHP < 7) and at other place we dont (`VisibilityRequiredFixer` `const` config for example). Since we do not do this the same everywhere I wonder what the best direction would be.